data-prepper-plugins/mutate-event/src/test/java/com/amazon/dataprepper/plugins/processor/mutateevent/CopyProcessorTests.java

```diff
@@ -0,0 +1,82 @@
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.record.Record;
+import org.junit.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@RunWith(MockitoJUnitRunner.class)
+public class CopyProcessorTests {
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private CopyProcessorConfig mockConfig;
+
+    @InjectMocks
+    private CopyProcessor processor;
+
+    @Test
+    public void testCopyProcessorTests() {
+        when(mockConfig.getFrom()).thenReturn("message");
+        when(mockConfig.getTo()).thenReturn("newMessage");
+        final Record<Event> record = getMessage("thisisamessage");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(true));
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo("thisisamessage"));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("thisisamessage"));
+    }
+
+    @Test
+    public void testCopyNoOverwriteProcessorTests() {
+        when(mockConfig.getFrom()).thenReturn("message");
+        when(mockConfig.getTo()).thenReturn("newMessage");
+        when(mockConfig.getSkipIfPresent()).thenReturn(true);
+        final Record<Event> record = getMessage("thisisamessage");
+        record.getData().put("newMessage", "test");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(true));
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo("test"));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("thisisamessage"));
+    }
+
+    private Record<Event> getMessage(String message) {
+        final Map<String, Object> testData = new HashMap();
+        testData.put("message", message);
+        return buildRecordWithEvent(testData);
+    }
+
+    private static Record<Event> buildRecordWithEvent(final Map<String, Object> data) {
+        return new Record<>(JacksonEvent.builder()
```

Comment:
Based on how you're using this, you can replace it by using  `Event event = JacksonEvent.fromMessage("this is a message")`;

Reply:
`fromMessage` is private. Should I make it public?

Reply:
Hm I didn't realize it was package private. I am not sure why that is. Maybe someone else that added can clarify the reasoning there. It seems like the intention for that function was to make it usable in plugin sources.